src/o365/spfx/commands/project/project-externalize/rules/PnPJSRule.spec.ts

```diff
@@ -0,0 +1,37 @@
+import * as assert from 'assert';
+import { Project } from '../../project-upgrade/model';
+import { PnPJsRule } from './PnPJsRule';
+
+describe('PnPJsRule', () => {
+  let rule: PnPJsRule;
+
+  beforeEach(() => {
+    rule = new PnPJsRule();
+  })
+
+  it('returns notification if dependecy is here', async () => {
+    const project: Project = {
+      path: '/usr/tmp',
+      packageJson: {
+        dependencies: {
+          '@pnp/pnpjs': '1.3.5'
+        }
+      }
+    };
+    const findings = await rule.visit(project);
+    assert.equal(findings.length, 1);
+  });
+
+  it('returns no notification if dependecy is not here', async () => {
+    const project: Project = {
+      path: '/usr/tmp',
+      packageJson: {
+        dependencies: {
+          '@pnp/pnpts': '1.3.5'
```

Comment:
Shouldn't this be `@pnp/pnpjs`?

Reply:
No it's on purpose, the idea is to test that it doesn't return anything when the dependency doesn't have a corresponding package json